proposals/simd/simdsignandunsignedextendedloads.md

```diff
@@ -0,0 +1,64 @@
+### **Proposal WebAssembly SIMD Modification**
+
+Currently as proposed there is an instructions defined in the WASM SIMD ISA as follows.
+
+**i8x16.mul** which is a register to register operation that takes 16 8 bit integers and
+
+multiplies them together resulting in an 8 bit value. If the distribution of the integers it flat this
+
+would result in a large percent of the instructions with overflow. This is a problem for many applications.
+
+
+### Proposed new instructions
+
+Six new load instructions are being proposed to make integer multiplies easier. i8x16zxload, i8x16sxload, i16x8zxload, i16x8sxload, i32x4zxload, i32x4sxload. This would make i8, i16, i32 multiplies useful and more practical for applications such as machine learning, image compression and video and rendering data processing.The new instructions would take consecutive integers of the corresponding size and zero sign extend and sign extend the consecutive bytes, words or dword to the promoted size of signed or unsigned data. An example of zero sign extend is shown below:  Intel and ARM both have this capability by doing the following:
+
+Intel Instructions:
+
+
+
+*   movzxbw
+*   movzxwd
+*   movzxdq
+*   movsxbw
+*   movsxwd
+*   Movsxdq
+
+ARM Instructions:
+
+
+
+*   LDR X0, [X1] Load from the address in X1
+*   LDR X0, [X1, #8] Load from address X1 + 8
+*   LDR X0, [X1, X2] Load from address X1 + X2
+*   LDR X0, [X1, X2, LSL, #3] Load from address X1 + (X2 << 3)
+*   LDR X0, [X1, W2, SXTW] Load from address X1 + sign extend(W2)
+*   LDR X0, [X1, W2, SXTW, #3] Load from address X1 + (sign extend(W2) << 3)
+
+So the new instructions for WASM would be defined as follows:
+
+
+
+*   i8x8.zxload
+*   i16x4.zxload
+*   i32x2.zxload
+*   i8x8.sxload
+*   i16x4.sxload
+*   i32x2.sxload
```

Comment:
All these types are 64-bit wide vector types, but the spec does not define `v64` anywhere.

Reply:
I find the document and the issue quite unclear, so I might be completely misunderstanding what problem this is trying to address, but if what you want is to be able load, e.g., 4 x 16-bit integers into a `v128` containing 4 x 32-bit integers by sign or zero extending them, then I would prefer something that's more similar to the scalar intrinsics, e.g., like `i32x4.load_i16x4_s(memarg) -> v128`.

Reply:
I think the intent here is to introduce something equivalent to the intel pmovzx(b/d/q) instruction, so the operation would be as follows - 

Packed_Zero_Extend_BYTE_to_WORD(DEST, SRC)
DEST[15:0] <- ZeroExtend(SRC[7:0]);
DEST[31:16] <- ZeroExtend(SRC[15:8]);
DEST[47:32] <- ZeroExtend(SRC[23:16]);
DEST[63:48] <- ZeroExtend(SRC[31:24]);
DEST[79:64] <- ZeroExtend(SRC[39:32]);
DEST[95:80] <- ZeroExtend(SRC[47:40]);
DEST[111:96] <- ZeroExtend(SRC[55:48]);
DEST[127:112] <- ZeroExtend(SRC[63:56]);

So it's not strictly a V128->V128 operation. I do agree that this would benefit from clearer names instead of zx/sx convention that's not very readable.